Specs/Scene/ModelExperimental/PointCloudStylingPipelineStageSpec.js

```diff
@@ -0,0 +1,629 @@
+import createScene from "../../createScene.js";
+import ShaderBuilderTester from "../../ShaderBuilderTester.js";
+
+describe(
+  "Scene/ModelExperimental/PointCloudStylingPipelineStage",
+  function () {
+    let scene;
+    const mockPrimitive = {
+      attributes: [
+        {
+          semantic: VertexAttributeSemantic.POSITION,
+          min: new Cartesian3(0, 0, 0),
+          max: new Cartesian3(1, 1, 1),
+          count: 64,
+        },
+      ],
+    };
+
+    const mockRuntimeNode = {
+      // prettier-ignore
+      transform: new Matrix4(2, 0, 0, 1,
+                             0, 2, 0, 0,
+                             0, 0, 2, 0,
+                             0, 0, 0, 1),
+    };
+
+    function mockGltfRenderResources(pointCloudShading) {
+      const attenuation = defined(pointCloudShading)
+        ? pointCloudShading.attenuation
+        : false;
+
+      const shaderBuilder = new ShaderBuilder();
+      const uniformMap = {};
+      const mockModel = {
+        type: ModelExperimentalType.GLTF,
+        pointCloudShading: pointCloudShading,
+        _attenuation: attenuation,
+      };
+
+      return {
+        shaderBuilder: shaderBuilder,
+        uniformMap: uniformMap,
+        runtimeNode: mockRuntimeNode,
+        model: mockModel,
+      };
+    }
+
+    function mockPntsRenderResources(options) {
+      const pointCloudShading = options.pointCloudShading;
+      const attenuation = defined(pointCloudShading)
+        ? pointCloudShading.attenuation
+        : false;
+
+      const shaderBuilder = new ShaderBuilder();
+      const uniformMap = {};
+      const mockModel = {
+        type: ModelExperimentalType.TILE_PNTS,
+        content: options.content,
+        style: options.style,
+        pointCloudShading: pointCloudShading,
+        _attenuation: attenuation,
+      };
+
+      return {
+        shaderBuilder: shaderBuilder,
+        uniformMap: uniformMap,
+        runtimeNode: mockRuntimeNode,
+        model: mockModel,
+        alphaOptions: {},
+      };
+    }
+
+    beforeAll(function () {
+      scene = createScene();
+    });
+
+    afterAll(function () {
+      scene.destroyForSpecs();
+    });
+
+    beforeEach(function () {
+      scene.morphTo3D(0.0);
+      scene.camera = new Camera(scene);
+      scene.renderForSpecs();
+    });
+
+    it("adds common uniform and code to the shader", function () {
+      const renderResources = mockGltfRenderResources();
+      const shaderBuilder = renderResources.shaderBuilder;
+      const uniformMap = renderResources.uniformMap;
+
+      PointCloudStylingPipelineStage.process(
+        renderResources,
+        mockPrimitive,
+        scene.frameState
+      );
+
+      ShaderBuilderTester.expectHasVertexDefines(shaderBuilder, []);
+      ShaderBuilderTester.expectHasFragmentDefines(shaderBuilder, []);
+
+      ShaderBuilderTester.expectHasVertexUniforms(shaderBuilder, [
+        "uniform vec4 model_pointCloudAttenuation;",
```

Comment:
Since we combined the styling/attenuation stages, is there a better name for this variable (while still being shorter than the original `u_pointSizeAndTimeAndGeometricErrorAndDepthMultiplier`?

maybe `model_pointCloudParameters` or something?